app/models/user.rb

```diff
@@ -5,8 +5,12 @@ class User < ActiveRecord::Base
 
   has_secure_password validations: false
 
-  has_many :reviews
-  has_many :queue_items, -> { order(:position) } # default is asc
+  has_many :reviews, -> { order(created_at: :desc) }
+  has_many :queue_items, -> { order(:position) } 
+  has_many :following_relationships, class_name: "Relationship", foreign_key: :follower_id
+  has_many :leading_relationships, class_name: "Relationship", foreign_key: :leader_id
+
+  before_create :generate_token
```

Comment:
I would not recommend generating token upon user creation. As mentioned above, its better to generate token for password reset when it is requested actually.